Source/DafnyLanguageServer.Test/Various/ConcurrentInteractionsTest.cs

```diff
@@ -75,36 +75,37 @@ decreases y
       }
 
       // Add a space in the document
-      var change1 = MakeChange(documentItem.Version + 1, new Range((0, 6), (0, 6)), " ");
-      client.DidChangeTextDocument(change1);
+      ApplyChange(ref documentItem, new Range((0, 6), (0, 6)), " ");
       // Save and don't wait, so the next save will interrupt and cancel verification
       client.SaveDocument(documentItem);
 
       // Remove the space, and use a non-consecutive version to test that the server doesn't drop the change
-      var change2 = MakeChange(documentItem.Version + 4, new Range((0, 6), (0, 7)), "");
+      var change2 = MakeChange(documentItem.Version + 2, new Range((0, 6), (0, 7)), "");
+      documentItem = documentItem with { Version = change2.TextDocument.Version };
       client.DidChangeTextDocument(change2);
       // Save and don't wait, so the next save will interrupt and cancel verification
       client.SaveDocument(documentItem);
 
       // Do the previous again a few times. This seems to be what it takes to guarantee cancelling verification.
-      client.DidChangeTextDocument(MakeChange(documentItem.Version + 5, new Range((0, 6), (0, 6)), " "));
+      ApplyChange(ref documentItem, new Range((0, 6), (0, 7)), " ");
```

Comment:
Good refactoring that will make it easier to maintain. Note that you can also remove the "new Range".

Reply:
I was surprised that this is possible. It works by implicitly converting the 2-tuples to Positions and then implicitly converting two Positions to a Range.